metals/src/main/scala/scala/meta/internal/metals/JavaFormattingProvider.scala

```diff
@@ -0,0 +1,230 @@
+package scala.meta.internal.metals
+
+import java.util
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+import scala.xml.Node
+
+import scala.meta._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+import scala.meta.{inputs => m}
+
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.jdt.core.ToolFactory
+import org.eclipse.jdt.core.formatter.CodeFormatter
+import org.eclipse.jdt.internal.formatter.DefaultCodeFormatterOptions
+import org.eclipse.jface.text.BadLocationException
+import org.eclipse.jface.text.Document
+import org.eclipse.text.edits.MalformedTreeException
```

Comment:
This seems not to work with JDK 8 :fearful: 
```
'/home/runner/work/metals/metals/tests/unit/target/e2e/implementation/on-usage' .
Exception in thread "pool-18-thread-19" java.lang.UnsupportedClassVersionError: org/eclipse/text/edits/MalformedTreeException has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Reply:
Probably connected to https://github.com/redhat-developer/vscode-java/issues/1543

Maybe we need to use an earlier version of jdt core?

Reply:
Damn - that didn't even occur to me.  It looks as though the other option: https://github.com/google/google-java-format also requires jdk11.
I guess we could use an older version, but then it wouldn't format new language features.
Or disable it if the JDK Metals is running under is <11?
Or use `ShellRunner#runJava` with different versions of the formatter depending on JDK being used?
I'll have a think - shame as it was an easy win - or so I thought.

Reply:
Looks like we can use `3.19.0` version for now maybe? We can decide on switching from JDK 8 later

Reply:
I think 3.25 might work - I'll try and go back further if not...
https://github.com/eclipse-mirrors/org.eclipse.jdt.core/commit/32abdfb9c5100e8c2ff7669768b7d7fcc83c7e23#diff-b2b8692cb3b6e958a605b1cb599cc6b55c0f7c1ae57961b0a105a31b8a9f74ef